test/Runtime/Inputs/custom_rr_abi_utilities.h

```diff
@@ -0,0 +1,111 @@
+#define NUM_REGS 30
+
+// Apply `macro` to "all" registers. Skip x18 since it's reserved, and x30 since
+// it's the link register.
```

Comment:
Do we need to worry about other `-ffixed-x??` and any thing like x29 being reserved for FP?

Reply:
We don't actually need to worry about the ones we're skipping, honestly; we'd just generate functions that don't ever get used. (The `x30` one might not actually work if it got called, mind.)

Reply:
I don't think it matters so much for `ALL_REGS` as it does for `FUNCTION_REGS`; I think even the ones we skip would be OK in the former. The latter has to be a little more careful. We don't need to worry about `-ffixed-x??`, I don't think, because this is just for testing.

Reply:
Right, this is just for testing, so unless someone is going to build the tests that way, it's fine.

For the actual runtime entrypoints, any entrypoint corresponding to a reserved register wouldn't be called, so it would be harmless other than a bit of wasted code space.

Reply:
Well, given that this _is_ about code size, that would be kinda nice to handle.  I don't see a good way to do this though as there is no macro that we have to check that and I am loathe to add a user controlled define for this.  I suppose that if you are truly concerned about space, LTO is a thing, and at that point the DCE _should_ kick in if you are statically linking.

Reply:
Sounds about right. If somebody really needs to squeeze every byte out of a dynamically-linked runtime with extra reserved registers, we can figure it out then.